pkg/cloud/amazon/s3_storage.go

```diff
@@ -512,95 +542,105 @@ func newClient(
 		defer sp.Finish()
 	}
 
-	opts := session.Options{}
+	var loadOptions []func(options *config.LoadOptions) error
+	addLoadOption := func(option config.LoadOptionsFunc) {
+		loadOptions = append(loadOptions, option)
+	}
 
-	httpClient, err := cloud.MakeHTTPClient(settings)
+	client, err := cloud.MakeHTTPClient(settings)
 	if err != nil {
 		return s3Client{}, "", err
 	}
-	opts.Config.HTTPClient = httpClient
-
-	if conf.endpoint != "" {
-		opts.Config.Endpoint = aws.String(conf.endpoint)
-		opts.Config.S3ForcePathStyle = aws.Bool(true)
-
-		if conf.region == "" {
```

Comment:
There is no default region in v2 of the SDK, if region is empty then there is logic below to infer the bucket's region before constructing the client.